AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Core/AWSAPICategoryPluginTests.swift

```diff
@@ -256,4 +256,67 @@ class AWSAPICategoryPluginTests: XCTestCase {
         }
         return count
     }
+
+    /// - Given: Datastore plugin
+    /// - When: Plugin is configured, started and cleared
+    /// - Then: Plugin's publisher should not received .finished and dataStorePublisher should not be nil
+
+    func testStorageEngineClear() throws {
+        let startExpectation = expectation(description: "Start Sync should be called with start")
+        let clearExpectation = expectation(description: "Clear should be called")
+
+        var count = 0
+        let storageEngine = MockStorageEngineBehavior()
+        storageEngine.responders[.startSync] = StartSyncResponder { _ in
+            count = self.expect(startExpectation, count, 1)
+        }
+        storageEngine.responders[.clear] = ClearResponder { _ in
+            count = self.expect(clearExpectation, count, 2)
+        }
+        let storageEngineBehaviorFactory: StorageEngineBehaviorFactory = {_, _, _, _, _, _  throws in
+            return storageEngine
+        }
+        let dataStorePublisher = DataStorePublisher()
+        let plugin = AWSDataStorePlugin(modelRegistration: TestModelRegistration(),
+                                        storageEngineBehaviorFactory: storageEngineBehaviorFactory,
+                                        dataStorePublisher: dataStorePublisher,
+                                        validAPIPluginKey: "MockAPICategoryPlugin",
+                                        validAuthPluginKey: "MockAuthCategoryPlugin")
+
+        let finishNotReceived = expectation(description: "Finish should not be received")
+        finishNotReceived.isInverted = true
+        _ = plugin.publisher.sink { completion in
+            switch completion {
+            case .finished:
+                print("Finish received")
+                finishNotReceived.fulfill()
+            case .failure(let error):
+                print("Error \(error)")
+            }
+        } receiveValue: { event in
+            print(event)
```

Comment:
nit: perfer not printing, perhaps we can remove the print if it doesn't matter in this test or replace it with an assert